src/openrct2/core/Path.cpp

```diff
@@ -22,6 +22,31 @@
 
 #include <iterator>
 
+#if defined(_MSC_VER)
+#    define HAVE_STD_FILESYSTEM 1
+#elif defined(__APPLE__)
+#    define HAVE_STD_FILESYSTEM 0
+#elif defined(__has_include)
+#    if __has_include(<filesystem>)
+#        define HAVE_STD_FILESYSTEM 1
+#    else
+#        define HAVE_STD_FILESYSTEM 0
+#    endif
```

Comment:
Why is this necessary? Which case is it trying to cover other than the two above? Just trying to understand

Reply:
This kind of got stale while I was doing it.

Reply:
I have added comments. Which part is confusing?